kroxylicious-kms-provider-hashicorp-vault/src/main/java/io/kroxylicious/kms/provider/hashicorp/vault/config/JdkTls.java

```diff
@@ -0,0 +1,134 @@
+import java.security.SecureRandom;
+import java.security.cert.X509Certificate;
+import java.util.Optional;
+
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.TrustManagerFactory;
+import javax.net.ssl.X509TrustManager;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.kroxylicious.proxy.config.tls.InsecureTls;
+import io.kroxylicious.proxy.config.tls.Tls;
+import io.kroxylicious.proxy.config.tls.TrustProvider;
+import io.kroxylicious.proxy.config.tls.TrustProviderVisitor;
+import io.kroxylicious.proxy.config.tls.TrustStore;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * Builds a JDK SSLContext used for vault communicateion.
+ *
+ * @param tls tls configuration
+ *
+ * TODO ability to restrict by TLS protocol and cipher suite.
+ */
+public record JdkTls(Tls tls) {
+
+    private static final Logger logger = LoggerFactory.getLogger(JdkTls.class);
+
+    public JdkTls {
+        if (tls != null && tls.key() != null) {
+            logger.warn("TLS key material is currently not supported by the vault client");
+        }
+    }
+
+    public static final X509TrustManager INSECURE_TRUST_MANAGER = new X509TrustManager() {
+
+        // suppressing sonar security warning as we are intentionally throwing security out the window
+        @SuppressWarnings("java:S4830")
+        @Override
+        public void checkClientTrusted(X509Certificate[] chain, String authType) {
+            // do nothing, the api is to throw if not trusted
+        }
+
+        // suppressing sonar security warning as we are intentionally throwing security out the window
+        @SuppressWarnings("java:S4830")
+        @Override
+        public void checkServerTrusted(X509Certificate[] chain, String authType) {
+            // do nothing, the api is to throw if not trusted
+        }
+
+        @Override
+        public X509Certificate[] getAcceptedIssuers() {
+            return new X509Certificate[0];
+        }
+    };
+    public static final TrustManager[] INSECURE_TRUST_MANAGERS = { INSECURE_TRUST_MANAGER };
+
+    public SSLContext sslContext() {
+        try {
+            if (tls != null && tls.trust() != null) {
+                return customSSLContext(tls.trust());
+            }
+            else {
+                return SSLContext.getDefault();
+            }
+        }
+        catch (Exception e) {
+            throw new SslConfigurationException(e);
+        }
+    }
+
+    @NonNull
+    private SSLContext customSSLContext(TrustProvider trust) throws NoSuchAlgorithmException, KeyManagementException {
+        Optional<TrustManager[]> trustManagers = getTrustManagers(trust);
+        return trustManagers.map(JdkTls::getSslContext).orElse(SSLContext.getDefault());
+    }
+
+    @NonNull
+    private static SSLContext getSslContext(TrustManager[] trustManagers1) {
+        try {
+            SSLContext rsa = SSLContext.getInstance("TLS");
```

Comment:
Let's respect the environment's default protocol name.   
```suggestion
            SSLContext rsa = SSLContext.getInstance(SSLContext.getDefault().getProtocol());
```